Plugin/Magento/Framework/View/Element/UiComponent/DataProvider/RegularFilterPlugin.php

```diff
@@ -58,20 +64,96 @@ public function aroundApply(
         Collection $collection,
         Filter $filter
     ) {
+        $filterValue = $filter->getValue();
```

Comment:
@garmashdev This would causes an error "call getValue on null" in some cases

Reply:
Or some instances does not support getValue method

Reply:
@dumega 
$filter it is required method parameter and have strong type declaration
Magento\Framework\Api\Filter\Filter
```
    public function aroundApply(
        RegularFilter $subject,
        callable $proceed,
        Collection $collection,
        Filter $filter
    ) {
        if ($collection instanceof OrderGridCollection &&
```

Also $filter->getValue() uses in Subject class in magento core without any checks:
Magento\Framework\View\Element\UiComponent\DataProvider\RegularFilter
```
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento\Framework\View\Element\UiComponent\DataProvider;

use Magento\Framework\Data\Collection;
use Magento\Framework\Api\Filter;

/**
 * Class RegularFilter
 */
class RegularFilter implements FilterApplierInterface
{
    /**
     * Apply regular filters like collection filters
     *
     * @param Collection $collection
     * @param Filter $filter
     * @return void
     */
    public function apply(Collection $collection, Filter $filter)
    {
        $collection->addFieldToFilter($filter->getField(), [$filter->getConditionType() => $filter->getValue()]);
    }
}
```

Reply:
@garmashdev It's no harm to check if the filter is set because I can imagine that some 3party modules can break this logic.

Let's also wrap up all code into try {} catch block - just in case.

Reply:
updated